plugins/woocommerce/src/Internal/ComingSoon/ComingSoonRequestHandler.php

```diff
@@ -195,6 +197,36 @@ public function experimental_filter_theme_json_theme( $theme_json ) {
 		$theme_data = $theme_json->get_data();
 		$font_data  = $theme_data['settings']['typography']['fontFamilies']['theme'] ?? array();
 
+		// Check if the current theme is a child theme. And if so, merge the parent theme fonts with the existing fonts.
+		if ( wp_get_theme()->parent() ) {
+			$parent_theme           = wp_get_theme()->parent();
+			$parent_theme_json_file = $parent_theme->get_file_path( 'theme.json' );
+
+			if ( is_readable( $parent_theme_json_file ) ) {
+				$parent_theme_json_data = json_decode( file_get_contents( $parent_theme_json_file ), true ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
+
+				if ( isset( $parent_theme_json_data['settings']['typography']['fontFamilies'] ) ) {
+					$parent_fonts = $parent_theme_json_data['settings']['typography']['fontFamilies'];
+
+					// Merge parent theme fonts with existing fonts.
+					foreach ( $parent_fonts as $parent_font ) {
+						$found = false;
+						foreach ( $font_data as $existing_font ) {
+							if ( isset( $parent_font['name'] ) && isset( $existing_font['name'] ) &&
+							$parent_font['name'] === $existing_font['name'] ) {
+								$found = true;
+								break;
+							}
+						}
+
+						if ( ! $found ) {
+							$font_data[] = $parent_font;
+						}
+					}
```

Comment:
What do you think about adding parent theme fonts to the `$fonts_to_add` array (defined just below this code) instead? It looks like we then loop through them (after adding inter and cardo) in a similar fashion to the code you have here.